libtransmission/handshake.cc

```diff
@@ -247,7 +249,7 @@ static handshake_parse_err_t parseHandshake(tr_handshake* handshake, struct evbu
     handshake->peer_id = peer_id;
 
     /* peer id */
-    tr_logAddTraceHand(handshake, "peer-id is [%" TR_PRIsv "]", TR_PRIsv_ARG(peer_id));
+    tr_logAddTraceHand(handshake, fmt::format("peer-id is '{}'", std::data(peer_id)));
```

Comment:
Not sure this will print well, also may not even be null-terminated. Maybe a custom formatter for the type, unless fmtlib already has one built-in for `std::array<>` and its output suits us?

Reply:
Good catch on the array-bounds read error. 

There's not a builtin for array<char>, but we can wrap it in a std::string_view easily enough...

c9dee6ae8971e2acd482029f9a7928e40b0d20c3